### PR TITLE
OCM-11703 | Update provider flag check 

### PR DESCRIPTION
**Pull request by JakobGray**
Issue: https://issues.redhat.com/browse/OCM-11703

Require provider, do not default to AWS, and check for provider-specific flags

### CODE REVIEW

File: cmd/ocm/create/cluster/cmd.go [hunk 1]
```diff
@@ -912,6 +918,22 @@ func promptName(argv []string) error {
...
+func promptProvider(fs *pflag.FlagSet, connection *sdk.Connection) error {
+	// Only offer the 2 providers known to support OSD now;
+	// but don't validate if set, to not block `ocm` CLI from creating clusters on future providers.
+	providers, _ := osdProviderOptions(connection)
+
+	err := arguments.PromptOneOf(fs, "provider", providers)
+	if err != nil {
+		return err
+	}
+	if args.provider == "" {
```
Comment (ckandag) [inline cmd/ocm/create/cluster/cmd.go:931]: Given that provider is inferred for the case where subscription is gcpMarketPlace,
 i.e provider  is only required if subscription is not gcpMarketPlace,  
We cannot mark the provider flag as required (the Cmd.MarkFlagRequired)  and instead we are doing a  manual check  if !gcpMarketPlace

File: pkg/arguments/arguments.go [hunk 2]
```diff
@@ -157,6 +157,52 @@ func CheckIgnoredCCSFlags(ccs cluster.CCS) error {
...
+	}
+	if provider != cluster.ProviderAWS {
+		for _, flag := range awsExclusiveFlags {
+			if fs.Changed(flag) {
+				bad = append(bad, flag)
+			}
+		}
+	}
+	if len(bad) == 1 {
+		return fmt.Errorf("%s flag is meaningless using chosen provider", bad[0])
```
Comment (ckandag) [inline pkg/arguments/arguments.go:198]: shouldnt it be.  `flag is meaningless "when" using chosen provider`

**Comment by ckandag** [comment]
please review err msg , looks good otherwise

**Review by ckandag — APPROVED** [review]

File: pkg/arguments/arguments.go (hunk 2 above)
Comment (JakobGray) [reply pkg/arguments/arguments.go]: Updated verbiage

**Pull request merged**